### PR TITLE
Add test for in-browser verification

### PR DESCRIPTION
**Pull request by flaub**
The browser test for this requires:

```
cargo xtask install
cargo xtask gen-receipt
```

Afterwards, you can run the test with:
```
cd examples/browser-verify
npm install
npm test -- --$BROWSER
```

where `$BROWSER` can be one of:
* `chrome`
* `firefox`
* `safari`

### CODE REVIEW

**Comment by flaub** [comment]
This is based on #703

File: examples/browser-verify/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+[package]
+name = "rust-webpack-template"
+description = "My super awesome Rust, WebAssembly, and Webpack project!"
```
Comment (mothran) [inline examples/browser-verify/Cargo.toml:3]: Might be worth making this more fit to the project.

File: examples/browser-verify/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+[lib]
+crate-type = ["cdylib"]
+
+[features]
+# If you uncomment this line, it will enable `wee_alloc`:
+#default = ["wee_alloc"]
+
+[dependencies]
+# The `console_error_panic_hook` crate provides better debugging of panics by
```
Comment (mothran) [inline examples/browser-verify/Cargo.toml:15]: Are most of these comments from the webpack template? Should we include them in our example as well?

File: examples/browser-verify/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+npm test -- --firefox
+
+# Runs tests in Chrome
+npm test -- --chrome
+
+# Runs tests in Safari
+npm test -- --safari
+```
+
+## What does each file do?
```
Comment (mothran) [inline examples/browser-verify/README.md:34]: Should we make this README more zkvm / r0 focused vs webpack?

File: examples/browser-verify/package.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
+  "author": "You <you@example.com>",
```
Comment (mothran) [inline examples/browser-verify/package.json:2]: template values.

File: examples/browser-verify/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+// This is like the `main` function, except for JavaScript.
+#[wasm_bindgen(start)]
+pub fn main_js() -> Result<(), JsValue> {
+    // This provides better error messages in debug mode.
+    // It's disabled in release mode so it doesn't bloat up the file size.
+    #[cfg(debug_assertions)]
+    console_error_panic_hook::set_once();
+
+    // Your code goes here!
+    console::log_1(&JsValue::from_str("Hello world!"));
```
Comment (mothran) [inline examples/browser-verify/src/lib.rs:35]: Still TODO here right?

File: examples/browser-verify/README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
+## How to install
```
Comment (tzerrell) [inline examples/browser-verify/README.md:1]: I created a PR that converts your PR comments into a basic readme: https://github.com/risc0/risc0/pull/745

Comment (flaub) [reply examples/browser-verify/README.md]: You're the best!

**Review by tzerrell — COMMENTED** [review]
An exciting example to add! Looking good to me modulo boilerplate cleanup.

**Review by SchmErik — COMMENTED** [review]
looks good! but I have a question: one of the things we wanted to do with the verifier was to build it in a no_std environment because some projects run the verifier on a no_std wasm environment. Does this PR take care of these breakages?

**Comment by flaub** [comment]
> looks good! but I have a question: one of the things we wanted to do with the verifier was to build it in a no_std environment because some projects run the verifier on a no_std wasm environment. Does this PR take care of these breakages?

This PR proves that running a verify within the browser works, but it doesn't specifically check that `no_std` works because WASM does have std support, it's just mostly stubbed out.

**Comment by flaub** [comment]
Are we OK to land this or is there anything else that needs to be addressed?

**Comment by mothran** [comment]
> Are we OK to land this or is there anything else that needs to be addressed?

I have a few small comments left about the left over default values / comments in places that would be nice to clean before merge but I am happy with the content!

**Review by mothran — APPROVED** [review]

**Pull request merged**